core/net/rpl/rpl-icmp6.c

```diff
@@ -887,6 +887,7 @@ dao_input_nonstoring(void)
   }
 
   if(flags & RPL_DAO_K_FLAG) {
+    uip_clear_buf();
     dao_ack_output(instance, &dao_sender_addr, sequence);
```

Comment:
Yes I would definitely add it there as well, even though we don't see a problem now, a bug might arise with future modifications!